viewer/Viewer.h

```diff
@@ -111,23 +123,65 @@ namespace Enki
 		typedef QMapIterator<const std::type_info*, const std::type_info*> ManagedObjectsAliasesMapIterator;
 		ManagedObjectsAliasesMap managedObjectsAliases;
 		
+		typedef std::pair<QString, unsigned int> ViewerErrorMessage;
+		std::list<ViewerErrorMessage> messageList;
+		QString controlError1;	// dislpayed if user try to move an object in trackball mode : action forbiden to prevent glitchies
+		QString controlError2;	// dislpayed if user try to move camera in trackball mode
+		QString controlHelp;	// dislpayed if user press F1
+
+		struct ExtendedAttributes
+		{
+			bool movableByPicking;
+
+			ExtendedAttributes():movableByPicking(false){};
+		};
+		std::map<PhysicalObject*, ExtendedAttributes> objectExtendedAttributesList;
+
 		bool mouseGrabbed;
 		QPoint mouseGrabPos;
 		double wallsHeight;
-		
-		bool doDumpFrames;
+		//! to know if camera is in trackball mode
+		bool trackballView;
 		int dumpFramesCounter;
 	
+		PhysicalObject *pointedObject, *selectedObject;
+		QVector3D pointedPoint;
+		bool movingObject;
+
 	public:
 		ViewerWidget(World *world, QWidget *parent = 0);
 		~ViewerWidget();
 	
+		void addManagedObjectsAlias(const std::type_info* key, const std::type_info* value);
+
+		World* getWorld();
+		QVector3D getPointedPoint() const;
+		PhysicalObject* getPointedObject();
+		PhysicalObject* getSelectedObject();
+		bool isTrackballActivated() const;
+		QString getHelpString() const;
+		bool isMovableByPicking(PhysicalObject* object) const;
+
+		/*!
+			\brief Specify if an object is movable by picking.
+			By default all object are not movable, so it's prefered to only specify object wich are mouvable by picking.
+			This function add an extended attribute to an object using an association table, so it's to the user responsibility to
+			delete the extended parameter if object is remove to the world, using the "removeExtendedAttributes" function
+		*/
+		void setMovableByPicking(PhysicalObject* object, bool movable);
+		void removeExtendedAttributes(PhysicalObject* object);
+
+		void keyPressEvent(QKeyEvent* event);
```

Comment:
You're right it's need to be in public slot.
Maybe this is the origin of the qt call problem !
But I can't test it now : playground actually inherit from viewer, so the function overload has to call the viewer keyPressed function (like now).
I will test it when playground will be a class inherited from mainwindow